SmithyDafnyMakefile.mk

```diff
@@ -525,6 +525,7 @@ transpile_test_java: _transpile_test_all _mv_test_java
 # To avoid `java/implementation-java` the code is generated and then moved.
 _mv_implementation_java:
 	rm -rf runtimes/java/src/main/dafny-generated
+	mkdir -p runtimes/java/src/main
```

Comment:
Is this right?
If you run `mkdir` against an existing directory, does it succeed?
Or does it fail?

Reply:
It succeeds - `-p` makes it recursively create all missing directories, so it's perfectly happy if all already exist

Reply:
Notice the same thing is already on the `runtimes/java/src/test` directory. I actually don't know how we got away without this fix for so long. :)

Reply:
Because you did not use to generate the template files,
and we had to manually stub the __default.java files in the Java extern source path.

Thus, we always had a directory.